### PR TITLE
fixings for xcode backend

### PR DESCRIPTION
**Pull request by nacho4d**
- Fix targets that use 1 precompiled header. It does not work for various precompiled headers yet. See https://github.com/mesonbuild/meson/issues/4242 
- Minor change: Sort some objects as xcode expects (improve xcodeproj correctness)
- Minor change: Some fixes to be able to run test cases in local machine and in CI in the future with Xcode

### CODE REVIEW

File: run_project_tests.py [hunk 1]
```diff
@@ -393,6 +393,7 @@ def _run_test(testdir, test_build_dir, install_dir, extra_args, compiler, backen
 
 def gather_tests(testdir: Path):
     tests = [t.name for t in testdir.glob('*')]
+    tests = [t for t in tests if t != '.DS_Store'] # Filter non-tests files (macOS Finder indexing files, etc)
```
Comment (jon-turney) [inline run_project_tests.py]: I think you could just filter out all dotfiles here (since `Path.glob()` doesn't)

Comment (nacho4d) [reply run_project_tests.py]: Done 👍

**Comment by nacho4d** [comment]
I have investigated and apparently **in Xcode** (because it uses `GCC_PREFIX_HEADER` build setting) it is possible to set **only one precompiled header per target**. 

Test case `13 pch` has one target that needs two pch: one for c files and one for cpp files; This is not supported in Xcode and this target is skipped. When doing `meson build --backend=xcode` for a project that has a target with 2 or more precompiled headers it will warn:

> Unsupported Xcode configuration: More than 1 precompiled header found "['header1.h', 'header2.h']". Target "foo" might not compile correctly.

One way to fully implement this would be to: 
1. create a custom build phase (shell script)
2. precompile as many as headers needed in this phase
3. pass the compiled header path to *each source file* with `-I` compiler flag.

Since Xcode has discouraged usage of pch in favor of modules since various versions ago I thought of implementing this when somebody needs/requires it. (Older versions used to create a default pch file, newer versions don't. Also have the option turned off by default)

Some references:
https://opensource.apple.com/source/clang/clang-425.0.24/src/tools/clang/docs/PCHInternals.html

**Comment by jpakkane** [comment]
Could we merge what we have already? Incremental fixing is better than trying to do one megacommit.

**Comment by nacho4d** [comment]
Yes. I agree. Current changes are ok to merge. :)

**Pull request merged**